### PR TITLE
NanoAODv7 code updates

### PR DESCRIPTION
**Pull request by peruzzim**
#### PR description:

This PR includes updates for NanoAODv7.
No other workflows are expected to be modified.

Summary of changes:

- updated betastar variables for jets
- fixed gen-photon matching
- updated b-jet and c-jet energy regressions
- consolidate electron trigger object filter bit definitions across years
- added filter bits for high pt electron and muon triggers
- updated muon selectors
- minor fixes and additional variables in fat jets
- added isolated gen-level photons
- fixes to event counters and weight sums
- fix muon IP sign
- updated photon IDs (esp. added Fall17v2 to 2016)
- JetMET info additions (unclustered MET, PUPPI MET total uncertainties)
- added SC energy for electrons
- critical fixes to gen weights storage in randomized parameter scans
- additional LHE info for MG reweighting
- additions to trigger objects (esp. for b-tagged jet paths)
- updated PU jet ID with 94X and 102X trainings
- fixed PSWeight scaling

### CODE REVIEW

**Comment by peruzzim** [comment]
please test workflow 1325.6,1325.7,1325.8,1329.1,136.7722,136.7952,136.8521 with cms-data/PhysicsTools-NanoAOD#12

**Comment by peruzzim** [comment]
+xpog

changes in NanoAOD workflows only

**Comment by perrotta** [comment]
+1
- Differences with respect to #29179 (already signed by reco) are only in the nanoAOD part of the code. Therefore, the same comments can be repeated here
- Changes in reco code refer to
   - PhysicsTools/PatAlgos/plugins/LeptonUpdater.cc, where two configuration parameters needed to fix the sign of the impact parameter were added: their default settings do not affect reco outputs
   - Modifications in RecoJets/JetProducers/plugins/BetaStarVarProducer.cc only affect nanoAOD, which is the only user of the "BetaStarPackedCandidateVarProducer" by now
- Jenkins tests show no differences in reco outputs; I assume that changes observed in the nanoAOD outputs have been implicitly considered validated by @peruzzim, while [signing](https://github.com/cms-sw/cmssw/pull/29212#issuecomment-600132250) this PR for xpog

**Comment by perrotta** [comment]
@peruzzim may I suggest to close  #29179 and have only this one to be merged?

**Comment by santocch** [comment]
+1

**Comment by silviodonato** [comment]
waiting for cms-data/PhysicsTools-NanoAOD#12

**Comment by smuzaffar** [comment]
@silviodonato , as analysis abd xpog has signed this so i guess we can go ahead and merge the data pr

**Comment by silviodonato** [comment]
+1

File: PhysicsTools/NanoAOD/python/jets_cff.py [hunk 1]
```diff
@@ -223,8 +229,10 @@
         chEmEF = Var("chargedEmEnergyFraction()", float, doc="charged Electromagnetic Energy Fraction", precision= 6),
         neEmEF = Var("neutralEmEnergyFraction()", float, doc="neutral Electromagnetic Energy Fraction", precision= 6),
         muEF = Var("muonEnergyFraction()", float, doc="muon Energy Fraction", precision= 6),
-        jercCHPUF = Var("userFloat('jercCHPUF')", float, doc="Pileup Charged Hadron Energy Fraction with the JERC group definition", precision= 6),
```
Comment (makortel) [inline PhysicsTools/NanoAOD/python/jets_cff.py]: This removal broke 25202.01, I opened an issue https://github.com/cms-sw/cmssw/issues/29245.

**Pull request merged**